src/dsp_tools/commands/xml_validate/reformat_input.py

```diff
@@ -42,21 +45,28 @@ def to_data_rdf(root: etree._Element) -> DataRDF:
 
 def _deserialise_all_resources(root: etree._Element) -> list[ResourceRDF]:
     all_res = []
+    restype_mapper = {
+        "annotation": KNORA_API.Annotation,
+        "region": KNORA_API.Region,
+        "link": KNORA_API.LinkObj,
+        "video-segment": KNORA_API.VideoSegment,
+        "audio-segment": KNORA_API.AudioSegment,
+    }
     for res in root.iterchildren():
         match res.tag:
             case "resource":
                 all_res.append(_deserialise_one_resource(res))
             case "annotation" | "region" | "link" | "video-segment" | "audio-segment":
-                all_res.append(_deserialise_dsp_base_resource(res))
+                all_res.append(_deserialise_dsp_base_resource(res, restype_mapper[res.tag]))
```

Comment:
have you tested in the debugger that this line is actually reachable?

Reply:
Yes `test_reformat_input.py` Checks that the number of resources that were extracted is correct. And `test_xml_validate.py` Checks that the transformation was done correctly.